options/db_options.cc

```diff
@@ -284,6 +284,10 @@ static std::unordered_map<std::string, OptionTypeInfo>
          {offsetof(struct ImmutableDBOptions, fail_if_options_file_error),
           OptionType::kBoolean, OptionVerificationType::kNormal,
           OptionTypeFlags::kNone}},
+        {"use_options_file",
+         {offsetof(struct ImmutableDBOptions, use_options_file),
+          OptionType::kBoolean, OptionVerificationType::kNormal,
+          OptionTypeFlags::kNone}},
```

Comment:
This option should probably be marked as kDontSerialize (so that it does not end up written to the options file) and kCompareNever (so it is not checked when comparing options)